### PR TITLE
JavaScript capstone project - API-based webapp

### PR DESCRIPTION
**Pull request by Sboursen**
## Description

> **CarCollectors** is website made for users that are interested in renting a car. It allows the visitor to browse an extensive collection of beautiful cars using the [Cars API](https://carsapi1.docs.apiary.io/#reference/0/cars-collection). It also allows the user to like, comment and make a reservation for a given car. All data are saved thanks the [Involvement API](https://www.notion.so/Involvement-API-869e60b5ad104603aa6db59e08150270).

### General requirements

- [x] There are no linter errors.
- [x] Followed the **Github flow**.
- [x] The documentation is professional.

### HTML/CSS and JavaScript requirements

- [x] Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- [x] Followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)

### Milestones

- [x] Milestone 1: **Project Setup and topic.**
- [x] Milestone 2: **Creation of the Home interface. Creation of the comment and the reservation pop-ups**
- [x] Milestone 3: **Creation of the communication interfaces with the APIs**
- [x] Milestone 4: **Implementation of the likes, comments and the reservations functionalities**
- [x] Milestone 5: **Writing tests to ensure the robustness of the application**

### Additional notes

> We apologize for the submission delays. It was due to some difficulties handling the **Cars API**.

> We made a [video demo](https://drive.google.com/file/d/1cjoWow9qx2MITazNmDSKM9TQczIT64aw/view?usp=sharing) for our website.

> We easy access to our website please use this [Live link](https://sboursen.github.io/MV-Module2-capstone/)

**_Thank you for taking the time to review this pull request._**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1 +1,43 @@
...
+  HomeApplication,
+  getAllCarsCount,
+} from './modules/home/home-app';
+import { showCommentModal } from './modules/comments/comments-app';
+import showReservationModal from './modules/reservations/reservation-app';
+import Icon from './images/logo.png';
+
+const homeApp = new HomeApplication();
+
+function registerHomeEvents() {
```
Comment (mikethreels) [inline src/index.js]: - [x] according to the requirements the Project should use ES6, which means you should avoid usage of the function keyword, please use arrow functions instead.

File: src/modules/comments/comments-app.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    .addComment(commentObj)
+    .then(() => api.getComment(id))
+    .then((data) => {
+      displayComment(data);
+      displayCommentCounter(data);
+    });
+  form.reset();
+};
+
+export async function showCommentModal(e) {
```
Comment (mikethreels) [inline src/modules/comments/comments-app.js]: - [x] according to the requirements the Project should use ES6, which means you should avoid usage of the function keyword, please use arrow functions instead.

File: src/modules/home/home-app.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        } else {
+          numberOfLikes = numberOfLikes.likes;
+        }
+        like.textContent = numberOfLikes;
+      });
+    });
+  };
+}
+
+export function getAllCarsCount() {
```
Comment (mikethreels) [inline src/modules/home/home-app.js]: - [x] according to the requirements the Project should use ES6, which means you should avoid usage of the function keyword, please use arrow functions instead.

File: src/modules/home/home-utils.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      <div class="car-buttons flex flex-col gap-3 p-2 ">
+        <button id="model" class="comment-button px-2 py-1 border-2 hover:border-2 hover:shadow-sm hover:text-white hover:bg-sky-500"
+          type="button">Comment</button>
+        <button
+          class="reservation-button px-2 py-1 border-2 hover:border-2 hover:shadow-sm hover:text-white hover:bg-sky-500"
+          type="button">Reservation</button>
+      </div>
+    </div>`;
+
+export function fileExists(imageUrl) {
```
Comment (mikethreels) [inline src/modules/home/home-utils.js]: - [x] according to the requirements the Project should use ES6, which means you should avoid usage of the function keyword, please use arrow functions instead.

File: src/modules/reservations/reservation-app.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    date_end: endDateInput.value,
+  };
+
+  setReservedToAPI(dataBody).then(() => {
+    clearInputs();
+    getDataFromInvolvementAPI(id);
+  });
+};
+
+export default function showReservationModal(e) {
```
Comment (mikethreels) [inline src/modules/reservations/reservation-app.js]: - [x] according to the requirements the Project should use ES6, which means you should avoid usage of the function keyword, please use arrow functions instead.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job setting up the api calls and making them work 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._
- [x] The video should contain a highlight about a piece of the code you find interesting or are proud of making, please reshoot the video and include this

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- [x] [OPTIONAL] It would be nice to see all the authors of the project with their camera on in the video presentation.
- [ ] [OPTIONAL] The dislike is not persisting it would be nice if this is the case.
- [ ] [OPTIONAL] when adding a comment or a reservation if the fields are empty the user is still able to add it it would be nice if you add the functionality so that the user is required to fill in the fields before they are able to submit it.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Sboursen** [comment]
Hi @mikethreels, :wave: 

First, we would like to thank you for the extensive feedback. :pray: 

As for the requested changes: 

- [x] we have successfully changed all our functions to use only ES6 syntax (AKA arrow functions).

- [x] We added a new feature `The loader`, that is being shown when the cars are being fetched from the Cars API.

- [x] We made a new video, that shows us giving a demo of the project.

- [x] Unfortunately we can not make the dislike feature persistent, because there is now way to delete likes from the Involvement API.

- [x] We would like to leave the validation feature to next release as it's not required right now.

Best,

**Review by Lordkaito — APPROVED** [review]
Hi @Sboursen and @jerryowusu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Sboursen** [comment]
Hi @Lordkaito 👋 

Thank  you for the approval.

Best,

**Comment by jerryowusu** [comment]
Thanks a lot @Lordkaito for the swift response and approval. (we were a group of 3 though, which includes @mwafrika)

**Comment by jerryowusu** [comment]
Congratulations @Sboursen and @mwafrika, it wasn't easy but we sailed through


200w
.

**Comment by Sboursen** [comment]
Yes we did

**Comment by mwafrika** [comment]
Yes we did it for sure, @Sboursen  @Sboursen  and me congratulations 🥇 🥇 🥇 
It was not easy

**Pull request merged**